### PR TITLE
Remove pmm-sumo from codeowners

### PR DESCRIPTION
**Pull request by pmm-sumo**
**Description:** 

Remove @pmm-sumo from the list of codeowners as I'm no longer able to contribute enough time for OpenTelemetry Community (#12493)

This is also a followup from #12596 and a [suggestion](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/12596#issuecomment-1192589716) to break it into separate PR's for each component

### CODE REVIEW

**Comment by pmm-sumo** [comment]
(BTW, I'm no longer to apply "Skip Changelog" label and I think it would fit here)

File: .github/CODEOWNERS
Comment (TylerHelmuth) [inline .github/CODEOWNERS:67]: I support removing @pmm-sumo, but we've decided that all [code owners going forward need to be members of the OpenTelemetry organization](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md#becoming-a-code-owner), and there was not an exception made for vender-specific components.  Is there anyone else from Sumo that is already a member that could be added?

Comment (pmm-sumo) [reply .github/CODEOWNERS:67]: @swiatekm-sumo considering that you [already made several contributions](https://github.com/open-telemetry/opentelemetry-collector-contrib/pulls?q=is%3Apr+author%3Aswiatekm-sumo+is%3Aclosed) I would suggest to apply for membership as the preferred solution :)

Comment (pmm-sumo) [reply .github/CODEOWNERS]: @TylerHelmuth I haven't seen followup on my suggestion, so to avoid blocking this PR further, I removed this entry. I believe this was the only outstanding item

**Review by TylerHelmuth — APPROVED** [review]
Thanks again for all your contributions!

**Comment by pmm-sumo** [comment]
This is probably a simple change, wanted to bring it up again to avoid unnecessary confusion when assigning issues/etc. @open-telemetry/collector-contrib-maintainer :)

**Review by djaglowski — APPROVED** [review]
It's been great working with you, @pmm-sumo. Best of luck!

**Comment by pmm-sumo** [comment]
Thank you @djaglowski! Same here :)

**Pull request merged**
